rosette/rbl/rosette/produce.rbl

```diff
@@ -0,0 +1,172 @@
+;;; Copyright (c) 2018, RChain Cooperative
+;;; Author: Kent Shikama <kent@kentshikama.com>
+;;; This file is licensed under the Apache License, version 2.0.
+;;;
+;;; File containing the reflective method produce and its helper methods
+
+(defProc (fetch-channel-pattern-and-product chart channel
+                                            product-patterns channel-index channel-position)
+  (if (= channel-index channel-position)
+      [channel -1]
+      (let [[[data-list continuation-struct-list] (tbl-get chart channel)]]
+        (if (null? data-list)
+            []
+            (iterate loop [[[data & r] data-list] [data-index 0]]
+                     (if (match-pattern (tuple-safe-nth product-patterns channel-index) data)
+                         [channel data-index]
+                         (loop r (inc data-index))))))))
+
+(defProc (fetch-pattern-and-product chart channels product-patterns channel-position)
```

Comment:
I don't think it will either. The goal here is to get a reference implementation that can run Rholang example contracts. Now that @JosDenmark and @echatav are working on a rholang interpreter (that I assume doesn't need a tuplespace to run), maybe we need to have a discussion on what the goal of this RBL tuplespace implementation is. However, at least for now, this implementation allows hello_world_again.rho, token.rho, and (shortly - I've got a patch lined up) auction.rho to run on rholangweb.

There are a lot of optimizations that can be done here. 

1) For starters, https://rchain.atlassian.net/browse/ROS-340 .
2) If we are set on using an efficient ranking algorithm - for example "take the first available smallest number of channel match" - then we don't have to loop through to extract all the candidates and filter them.
3) Push more of the logic in the Tuplespace down a level - essentially write some methods as Rosette primitives.